### ISSUE TITLE
Directive formatting in output

### ISSUE THREAD

**Issue by acklavidian** [eemeli/yaml#112]
Is there anyway to have the `---` (directive maybe?) at the top of the document on stringification?'
eg:
```yaml
---
description: 'i do not know what that thing up there at the top is called'
possibly:
  - directive
  - triple dash
  - separator  
...
```

**Comment by eemeli** [comment]
At least for the moment the [directives-end](https://yaml.org/spec/1.2/spec.html#c-directives-end) indicator `---` is only included if the document has explicit `%YAML` or `%TAG` directives. Do you have a use case where it would be needed or useful in other cases?

**Comment by acklavidian** [comment]
Ah, directives-end indicator... thank you. Well I'm trying to parse and manipulate some yaml that exists in 3 dimensions. A ruby based CLI tool, a web based UI, and in version control. So the document gets written from this ruby tool and then interpreted by my web-UI code. Then my code (using your lib) overwrites the yaml files incase there were some changes. Then on top of that there is also version control watching these yaml files (git via bitbucket) So even if there isn't any changes my side will wanna remove the `---` at the top of the file and this in turn shows up in the version control. Obviously, this isn't a meaningful change. I know I could just prepend the `---\n` with string interpolation to the doc. I just would prefer not to do any string interpolation if your library can handle it cleaner.

**Comment by eemeli** [comment]
So the use case here is that you're trying to use the library together with a ruby tool that always includes the `---` indicator?

I'm pondering adding an option for this, given that it might be useful to have certainty about whether or not the indicator is included in the output. Or more likely reading its presence from the input, and if so including it in the input.

**Comment by acklavidian** [comment]
Yeah, that is pretty much the long and short of it. Either/Both of those options would work for my uses.

**Comment by eemeli** [comment]
Fixed in [version 1.6.0](../releases/tag/v1.6.0).
